streams/shared/src/main/scala/zio/stream/StreamEffect.scala

```diff
@@ -365,12 +365,10 @@ private[stream] object StreamEffect extends Serializable {
       }
     }
 
-  final def fromIterator[R, E, A](iterator: ZManaged[R, E, Iterator[A]]): StreamEffect[R, E, A] =
+  final def fromIterator[A](iterator: Iterator[A]): StreamEffect[Any, Nothing, A] =
```

Comment:
Can we implement `StreamEffect.fromIterable` in terms of this (by calling `Iterable#iterator`?